hcmanager/classification_filter.go

```diff
@@ -0,0 +1,264 @@
+
+	var comparedHost []string
+
+	for hostname, details := range validatedHost {
+		isHostValid := true
+		for _, value := range details {
+			isValid := false
+			cpu, CPUOK := value.(bmh.CPU)
+			if CPUOK {
+				if checkCPUCount(mgr, cpu, expectedHardwareprofile.Cpu) {
+					isValid = true
+				}
+			}
+
+			ram, RAMOK := value.(int64)
+			if RAMOK {
+				if checkRAM(mgr, ram, expectedHardwareprofile.Ram) {
+					isValid = true
+				}
+			}
+
+			nics, NICSOK := value.(int)
+			if NICSOK {
+				if checkNICS(mgr, nics, expectedHardwareprofile.Nic) {
+					isValid = true
+				}
+			}
+
+			disk, DISKOK := value.([]bmh.Storage)
+			if DISKOK {
+				if checkDiskDetails(mgr, disk, expectedHardwareprofile.Disk) {
+					isValid = true
+				}
+			}
+
+			if !isValid {
+				isHostValid = false
+				break
+			}
+
+		}
+
+		if isHostValid {
+			comparedHost = append(comparedHost, hostname)
+			mgr.Log.Info(hostname, " Matches profile ", ProfileName)
+
+		} else {
+			mgr.Log.Info(hostname, " Did not matches profile ", ProfileName)
+		}
+
+	}
+
+	return comparedHost
+
+}
+
+//checkCPUCount this function checks the CPU details for both min and max parameters
+func checkCPUCount(mgr HardwareClassificationManager, cpu bmh.CPU, expectedCPU *hwcc.Cpu) bool {
+
+	if (expectedCPU.MaximumCount > 0) && (expectedCPU.MinimumCount > 0) {
+
+		mgr.Log.Info("", "Provided Minimum count for CPU", expectedCPU.MinimumCount, " and fetched count ", cpu.Count)
+		mgr.Log.Info("", "Provided Maximum count for CPU", expectedCPU.MaximumCount, " and fetched count ", cpu.Count)
+		if (expectedCPU.MinimumCount > cpu.Count) || (expectedCPU.MaximumCount < cpu.Count) {
+			mgr.Log.Info("CPU MINMAX COUNT did not match")
+			return false
+		}
+
+	} else if expectedCPU.MaximumCount > 0 {
+
+		mgr.Log.Info("", "Provided Maximum count for CPU", expectedCPU.MaximumCount, " and fetched count ", cpu.Count)
+		if expectedCPU.MaximumCount < cpu.Count {
+
+			mgr.Log.Info("CPU MAX COUNT did not match")
+			return false
+		}
+
+	} else if expectedCPU.MinimumCount > 0 {
+
+		mgr.Log.Info("", "Provided Minimum count for CPU", expectedCPU.MinimumCount, " and fetched count ", cpu.Count)
+		if expectedCPU.MinimumCount > cpu.Count {
+			mgr.Log.Info("CPU MIN COUNT did not match")
+			return false
+		}
+
+	}
+
+	if (expectedCPU.MaximumSpeedMHz > 0) && (expectedCPU.MinimumSpeedMHz > 0) {
+
+		MinSpeed := bmh.ClockSpeed(expectedCPU.MinimumSpeedMHz)
+		MaxSpeed := bmh.ClockSpeed(expectedCPU.MaximumSpeedMHz)
+
+		mgr.Log.Info("", "Provided Minimum ClockSpeed for CPU", MinSpeed, " and fetched ClockSpeed ", cpu.ClockMegahertz)
+		mgr.Log.Info("", "Provided Maximum ClockSpeed for CPU", MaxSpeed, " and fetched ClockSpeed ", cpu.ClockMegahertz)
+		if (MinSpeed > cpu.ClockMegahertz) || (MaxSpeed < cpu.ClockMegahertz) {
+			mgr.Log.Info("CPU MINMAX ClockSpeed did not match")
+			return false
+		}
+	} else if expectedCPU.MaximumSpeedMHz > 0 {
+		MaxSpeed := bmh.ClockSpeed(float64(expectedCPU.MaximumSpeedMHz / 1000))
```

Comment:
This is also defined once at the start of the function

Reply:
Done